File: Math/find_repeating_and_missing.py
# 1. You are given an arrya of integers whose size is n.
# 2. The array contains elements from 1 to n. It is given that one of the numbers
# is missing and one of the numbers is repeated.
# 3. You have to print repeating and missing element respectively.

from operator import mul
from math import factorial

n = int(input())
arr = [int(i) for i in input().split(" ")]
s = sum(arr)
m = 1
for i in range(n):
    m *= arr[i]
sn = n*(n+1)//2
f = factorial(n)

x = int(((s-sn)*f)/(m-f))
y = int((m*x)/f)

print(y)
print(x)

File: Hash Table/FindRepeatingAndMissing.java
// 1. You are given an arrya of integers whose size is n.
// 2. The array contains elements from 1 to n. It is given that one of the numbers
// is missing and one of the numbers is repeated.
// 3. You have to print repeating and missing element respectively.

import java.util.*;
import java.io.*;

public class FindRepeatingAndMissing {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();

        if (n == 1) {
            System.out.println(-1);
            return;
        }

        int[] arr = new int[n];
        int[] hashT = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = s.nextInt();
            hashT[arr[i] - 1]++;
        }

        int missing = 0, repeating = 0;

        for (int i = 0; i < n; i++) {
            if (hashT[i] == 0) {
                missing = i + 1;
            }

            if (hashT[i] > 1) {
                repeating = i + 1;
            }
        }

        System.out.println(repeating);
        System.out.println(missing);
    }
}
